### PR TITLE
✨ Update API spec for controller implementation

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

This PR extends API with fixes for validation and object serialization, and additional required methods.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #286 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by alexander-demicev — APPROVED** [review]

File: api/v1alpha1/capiprovider_conditions.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package v1alpha1
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// GetConditions return CAPI Provider conditions.
+func (p *CAPIProvider) GetConditions() clusterv1.Conditions {
```
Comment (richardcase) [inline api/v1alpha1/capiprovider_conditions.go:8]: Should we also have `SetConditions`?

File: api/v1alpha1/types.go [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+import (
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+// ProviderKind is a CAPIProvider kind string.
+const ProviderKind = "CAPIProvider"
+
 // ProviderName defines the designated provider infrastructure provider name.
 type ProviderName string
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply api/v1alpha1/types.go:12]: A name within the v1alpha1/types.go may mean a variety of things.

File: charts/rancher-turtles/Chart.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ keywords:
 - provisioning
 dependencies:
   - name: cluster-api-operator
-    version: v0.5.1
+    version: v0.7.0
```
Comment (alexander-demicev) [inline charts/rancher-turtles/Chart.yaml:16]: can you also bump all operator manifests to `v1alpha2`? Can be done in a follow-up PR

**Review by alexander-demicev — DISMISSED** [review]

File: api/v1alpha1/capiprovider_types.go [hunk 4]
```diff
@@ -104,19 +105,23 @@ type WorkloadIdentityRef struct {
...
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-
 // CAPIProvider is the Schema for the CAPI Providers API.
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.type"
+// +kubebuilder:printcolumn:name="ProviderName",type="string",JSONPath=".spec.name"
+// +kubebuilder:printcolumn:name="InstalledVersion",type="string",JSONPath=".status.installedVersion"
```
Comment (furkatgofurov7) [inline api/v1alpha1/capiprovider_types.go:123]: Q: do we have `.installedVersion` field defined in the current API?

Comment (Danil-Grigorev) [reply api/v1alpha1/capiprovider_types.go:123]: Yes, this is a part of inlined status of the CAPI operator

File: api/v1alpha1/types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply api/v1alpha1/types.go]: Fixed

File: charts/rancher-turtles/Chart.yaml (hunk 3 above)
Comment (Danil-Grigorev) [reply charts/rancher-turtles/Chart.yaml:16]: I think we can do this in a followup PR. CAPI operator uses v1alpha2 as a store version, so submitting v1alpha1 is still stored as v1alpha2 and is not critical.

File: api/v1alpha1/capiprovider_conditions.go (hunk 1 above)
Comment (Danil-Grigorev) [reply api/v1alpha1/capiprovider_conditions.go:8]: Please check CAPI Operator PR. The plan is to use the generic interface from upstream, and it needs help reviewing. For now this GetConditions is only required to satisfy the interface in logic https://github.com/rancher-sandbox/rancher-turtles/pull/293/files#diff-ccaf660a8ccddd6d42cb87d6455fcab1ca110bf8a65a9c3fad9af8c15a89eaf9R90 I’d prefer to keep changes minimal.

**Review by furkatgofurov7 — DISMISSED** [review]

File: charts/rancher-turtles/Chart.yaml (hunk 3 above)
Comment (Danil-Grigorev) [reply charts/rancher-turtles/Chart.yaml]: Unfortunately there is a bug in v0.7.0 preventing us from using it. Reverted it to 0.6.0, as this one works in testing.

**Review by richardcase — APPROVED** [review]

File: api/v1alpha1/capiprovider_conditions.go (hunk 1 above)
Comment (furkatgofurov7) [reply api/v1alpha1/capiprovider_conditions.go:8]: which PR?

File: charts/rancher-turtles/Chart.yaml (hunk 3 above)
Comment (Danil-Grigorev) [reply charts/rancher-turtles/Chart.yaml]: Should be eventually fixed by https://github.com/kubernetes-sigs/cluster-api-operator/pull/352.

**Pull request merged**